tools/scripts/artifacts/utils.ts

```diff
@@ -12,51 +12,64 @@ export class Utils {
   public static readonly EMPTY_GITHUB_COMMENTS =
     'No snapshots of projects have been published (probably no project is affected)';
 
+  public static globProjectJSON(): string[] {
+    const projects = execSync(
+      `npx glob --cwd=${process.cwd()} --absolute --dot ./**/project.json`
+    )
+      .toString()
+      .trim();
+    return projects.split('\n').filter((projectJSON) => {
+      return (
+        !projectJSON.includes('dist') &&
+        !projectJSON.includes('node_modules') &&
+        !projectJSON.includes('.git')
+      );
+    });
+  }
+
   public static getAffectedNxProjects(
     base: string,
     nxProjectKind: NxProjectKind,
     task: TASK = TASK.MAIN_SNAPSHOT,
     version: Version = new Version(),
     scope: string = ''
   ): NxProject[] {
-    const affectedString = execSync(
-      `npx nx affected:${
-        nxProjectKind === NxProjectKind.Application ? 'apps' : 'libs'
-      } --base=${base} --plain`
-    ).toString();
+    let affectedProjects = [];
+    if (nxProjectKind === NxProjectKind.Application) {
+      affectedProjects = Utils.getListOfAllAffectedApps(base);
+    } else {
+      affectedProjects = Utils.getListOfAllAffectedLibs(base);
+    }
     console.log(
       `Affected ${
         nxProjectKind === NxProjectKind.Application ? 'apps' : 'libs'
-      }: ` + affectedString
+      }: ` + affectedProjects.toString()
     );
-    let affectedNames: string[] = affectedString
-      .trim()
-      .split(' ')
-      .filter((i) => i)
-      .filter((i) => !i.startsWith('api-'))
+    let filteredAffected: string[] = affectedProjects
+      .filter((project) => !project.includes('e2e'))
```

Comment:
I'm also doubting this bit a little. Shouldn't we define some convention so we look for stuff starting/ending with `e2e` instead of just containing it _somewhere_ in its path? 🤔 Similar to the `api-` thing below.